src/node.rs

```diff
@@ -680,4 +668,44 @@ mod test {
         node.entries_mut().push(KdlEntry::new_prop("x", 2));
         assert_eq!(&node["x"], &2.into())
     }
+
+    #[test]
+    fn insertion() {
+        let mut node = KdlNode::new("foo");
+        node.push("pos0");
+        node.insert("keyword", 6.0);
+        node.push("pos1");
+        assert_eq!(node.entries().len(), 3);
+
+        node.insert(0, "inserted0");
+        node.insert(1, "inserted1");
+        assert_eq!(node.entries().len(), 3);
+        assert_eq!(node[0], "inserted0".into());
+        assert_eq!(node[1], "inserted1".into());
```

Comment:
this is not what I would expect. [`Vec::insert`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert) actually shifts existing elements to the right, so I would expect this assertion to be `.len(), 5`, and for `"pos0"` to be accessible at `node[2]`.

Reply:
Ah! I see, I misunderstood the doc comment on `KdlNode::insert`. The way you've described seems more intuitive.

Reply:
Oh wait, it looks like the current expected behavior for `KdlNode::insert_impl` is to replace entries, even if using `NodeKey::Index`. In the doc comment, when it says "If an entry already exists with the same key", does "key" in this case mean any `NodeKey` or specifically `NodeKey::Key`?

Reply:
yeah, it's weird, and imo, it needs to be different depending on whether the key is a number or a property, tbh. I think I had hash keys in mind when I wrote this, but when you're dealing with array-like data, you expect things to shift right.